src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -170,7 +171,7 @@ final void untrackStatement(final Statement statement)
 
    final void markCommitStateDirty()
    {
-      if (isAutoCommit) {
+      if (isAutoCommit || isReadOnly) {
```

Comment:
at present, if auto commit false and connection is and stays read only and not returned within bypass window, it is validating before leasing out because lastAccss is not initialized in markCommitStateDirty